site/site.go

```diff
@@ -58,6 +58,21 @@ var (
 	oauthHTML string
 
 	oauthTemplate *htmltemplate.Template
+
+	//go:embed static/oauth2device.html
```

Comment:
this means all of these files will actually be embedded in the binary twice

- once here
- once [over here](https://github.com/coder/coder/blob/a185d3a2c339f2b66fb6f79e7b7c046a2887b110/site/site_embed.go#L11) because everything in static/ gets copied to out/ [by vite](https://github.com/coder/coder/blob/a185d3a2c339f2b66fb6f79e7b7c046a2887b110/site/vite.config.mts#L25)

the version copied by vite is practically useless since it will always be the unfilled template, but it will always be accessible at http://<coder domain>/oauth2device.html, and so on for the other similar html files.

maybe this means we should have a site/templates/ folder to put stuff like this in?